lib/Parse/Lexer.cpp

```diff
@@ -827,10 +827,15 @@ void Lexer::lexDollarIdent() {
     }
   }
 
-  // It's always an error to see a standalone $, and we reserve
-  // $nonNumeric for persistent bindings in the debugger.
-  if (CurPtr == tokStart + 1 || !isAllDigits) {
-    if (!isAllDigits && !LangOpts.EnableDollarIdentifiers)
+  // It's always an error to see a standalone $
+  if (CurPtr == tokStart + 1) {
+    diagnose(tokStart, diag::expected_dollar_numeric);
+    return formToken(tok::unknown, TokStart);
+  }
+
+  // We reserve $nonNumeric for persistent bindings in the debugger.
+  if(!isAllDigits) {
+    if (!LangOpts.EnableDollarIdentifiers)
```

Comment:
What config changes? Maybe you could point me in the right direction as I was not aware that is affected any configuration files.

The `EnableDollarIdentifiers` flag has been there before and from my understanding the only difference is that now a single "$" will be lexed as an `tok:unknown` and cause a diagnose instead of being a `tok::identifier`.